src/arrayMethodReduce.js

```diff
@@ -4,9 +4,18 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(callback, ...initialValue) {
+    let accumulator = initialValue.length > 0
```

Comment:
The ternary operator usage for checking the length of initialValue can lead to incorrect results if initialValue is undefined. You need to check if initialValue is not undefined before checking its length.